elf/arch-riscv.cc

```diff
@@ -812,6 +841,43 @@ static void shrink_section(Context<E> &ctx, InputSection<E> &isec, bool use_rvc)
         delta += 4;
       break;
     }
+    case R_RISCV_HI20: {
+      // This relocation refers to an LUI instruction containing the high
+      // 20-bits to be relocated to an absolute symbol address.
+      // In some cases, LUI can be removed or replaced by C.LUI.
+
+      // If it's a static linking, we skip this relaxation for now, as it's
+      // currently not working.
+      if (ctx.arg.is_static)
+        break;
+
+      const SectionFragmentRef<E> *frag_ref = nullptr;
+      if (isec.rel_fragments && isec.rel_fragments[frag_idx].idx == i)
+        frag_ref = &isec.rel_fragments[frag_idx++];
```

Comment:
You shouldn't write this inside a switch-case because frag_idx would be increment only when rel is R_RISCV_HI20. We should increment it regardless of relocation type.

Reply:
Thanks, done.